src/nvim/api/autocmd.c

```diff
@@ -315,8 +315,13 @@ Array nvim_get_autocmds(Dict(get_autocmds) *opts, Error *err)
 ///
 /// @param opts Optional Parameters:
 ///         - callback: (string|function)
-///             - (string): The name of the viml function to execute when triggering this autocmd
-///             - (function): The lua function to execute when triggering this autocmd
+///             - (string): The name of the Vimscript function to execute when this autocmd is
+///                         triggered.
+///             - (function): Lua function to call when triggering this autocmd.
+///             - The function is passed a dict or table with the keys "match", "file", and "buf"
+///               with values corresponding to the expanded values of |<amatch>|, |<afile>|, and
+///               |<abuf>|, respectively. If the callback function returns |v:true|, then the
```

Comment:
I'd prefer if we minimized vimscript references in the API docs. `true` is sufficient here.

Reply:
The use of `v:true` was intentional: Vimscript functions must return `v:true` explicitly, not any “true” value. Saying it must return `true` is ambiguous, because in Vimscript `1` is true, but does not remove the autocommand.

It is less ambiguous to infer that `v:true` == `true` in Lua than the other way around.

Reply:
I feel this should be more explicit then. E.g. "note: for vimscript v:true must be returned".

Reply:
To add, nowhere else in the API docs do we make this distinction.

Reply:
Never hurts to be explicit.

Reply:
Do we not also need to do this for `nvim_buf_attach` and `nvim_set_decoration_provider`?

Reply:
Both of those only take Lua callbacks, so it's not an issue there.